h/services/annotation_moderation.py

```diff
@@ -15,12 +18,53 @@ def all_hidden(self, annotation_ids):
         if not annotation_ids:
             return set()
 
+        # TODO, move to the new column after migration and backfill migration
         query = self._session.query(AnnotationModeration.annotation_id).filter(
             AnnotationModeration.annotation_id.in_(annotation_ids)
         )
-
         return {m.annotation_id for m in query}
 
+    def set_status(
+        self, annotation: Annotation, user: User, status: ModerationStatus | None
```

Comment:
Here we set an status (almost) unconditionally.

We log it on the moderation_log table as well.